extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelProcessor.java

```diff
@@ -399,15 +395,19 @@ public List<CamelRoutesBuilderClassBuildItem> discoverRoutesBuilderClassNames(
     @Record(ExecutionTime.STATIC_INIT)
     @BuildStep
     CamelComponentNameResolverBuildItem componentNameResolver(
-            BuildTimeCamelCatalogBuildItem camelCatalog,
```

Comment:
It seems that `BuildTimeCamelCatalog` no other usage if we remove here. Just curious, what's the original idea about it?

Reply:
It is used elsewhere. To figure out which types can be made unremovable, just in case folks want to use CDI to wire up components etc.

Reply:
OK, I get it. Thanks!